src/main/java/org/prebid/server/auction/BidResponseCreator.java

```diff
@@ -514,12 +557,14 @@ private BidResponse toBidResponse(
                 toExtBidResponse(bidderResponses, bidRequest, cacheResult, videoStoredDataResult,
                         debugEnabled, bidErrors);
 
-        return BidResponse.builder()
+        BidResponse response = BidResponse.builder()
                 .id(bidRequest.getId())
                 .cur(bidRequest.getCur().get(0))
                 .seatbid(seatBids)
                 .ext(mapper.mapper().valueToTree(extBidResponse))
                 .build();
+        overwriteBidIdConsumer.accept(response);
```

Comment:
I'd probably just added method `toSeatbid`, which create appropriate objects. 
And modify `toSeatBid`